diff -dura dante-1.1.19/lib/clientprotocol.c dante-1.1.19.patch//lib/clientprotocol.c
--- dante-1.1.19/lib/clientprotocol.c	2005-12-29 06:17:24.000000000 -0500
+++ dante-1.1.19.patch//lib/clientprotocol.c	2010-10-30 15:15:27.306395005 -0400
@@ -160,10 +160,10 @@
 								 + sizeof(response->reply)
 								 ];
 			char *p = responsemem;
-
-			if (readn(s, responsemem, sizeof(responsemem), response->auth)
-			!= sizeof(responsemem)) {
-				swarn("%s: readn()", function);
+			int i = readn(s, responsemem, sizeof(responsemem),
+					response->auth);
+			if (i != sizeof(responsemem)) {
+				swarn("%s: readn() %d", function, i);
 				return -1;
 			}
 
@@ -200,10 +200,10 @@
 								+ sizeof(response->flag)
 								];
 			char *p = responsemem;
-
-			if (readn(s, responsemem, sizeof(responsemem), response->auth)
-			!= sizeof(responsemem)) {
-				swarn("%s: readn()", function);
+			int i = readn(s, responsemem, sizeof(responsemem),
+					response->auth);
+			if ( i != sizeof(responsemem)) {
+				swarn("%s: readn() %d", function, i);
 				return -1;
 			}
 
Only in dante-1.1.19.patch//lib: clientprotocol.c~
